nsxt/resource_nsxt_policy_ip_block_quota_test.go

```diff
@@ -0,0 +1,254 @@
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy IpBlockQuota resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy IpBlockQuota resource ID not set in resources")
+		}
+
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("Policy IpBlockQuota %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtPolicyIpBlockQuotaCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_ip_block_quota" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("Policy IpBlockQuota %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyIpBlockQuotaTemplate(createFlow bool, withContext bool) string {
+	var attrMap map[string]string
+	var ipblock string
+	if createFlow {
+		attrMap = accTestPolicyIpBlockQuotaCreateAttributes
+		ipblock = "test-ipv4"
+	} else {
+		attrMap = accTestPolicyIpBlockQuotaUpdateAttributes
+		ipblock = "test-ipv6"
+	}
+	context := ""
+	if withContext {
+		context = testAccNsxtPolicyMultitenancyContext()
+	}
+	return fmt.Sprintf(`
+resource "nsxt_policy_ip_block" "test-ipv4" {
+  %s
+  display_name = "%s-ipv4"
+  cidr         = "20.20.20.0/24"
+}
+
+resource "nsxt_policy_ip_block" "test-ipv6" {
+  %s
+  display_name = "%s-ipv6"
+  cidr         = "fc7e:0011:da43::0/48"
+}
+
+resource "nsxt_policy_ip_block_quota" "test" {
+  %s
+  display_name = "%s"
+  description  = "%s"
+
+  quota {
+    ip_block_paths        = [nsxt_policy_ip_block.%s.path]
+    ip_block_visibility   = "%s"
+    ip_block_address_type = "%s"
+    
+    single_ip_cidrs = %s
+    other_cidrs {
+      mask        = "%s"
+      total_count = %s
+    }
+  }
+
+}`, context, attrMap["display_name"], context, attrMap["display_name"], context, attrMap["display_name"], attrMap["description"], ipblock, attrMap["ip_block_visibility"], attrMap["ip_block_address_type"], attrMap["single_ip_cidrs"], attrMap["mask"], attrMap["total_count"])
+}
+
+func testAccNsxtPolicyIpBlockQuotaMinimalistic(withContext bool) string {
+	attrMap := accTestPolicyIpBlockQuotaCreateAttributes
+	context := ""
+	if withContext {
+		context = testAccNsxtPolicyMultitenancyContext()
+	}
+	return fmt.Sprintf(`
+resource "nsxt_policy_ip_block_quota" "test" {
+  %s
+  display_name = "%s"
+  quota {
+    ip_block_visibility   = "%s"
+    ip_block_address_type = "%s"
+    single_ip_cidrs       = %s
+
+    other_cidrs {
```

Comment:
Is it mandatory to specify the empty struct, or are you verifying that it won't throw a nil pointer exception when used?

Reply:
I've made `other_cidrs` required since its auto-assigned, hence must be either `Required` or `Computed`. The latter option causes non-intuitive behavior when block is deleted (instead of reverting to default, the quota would stay as its last value, since terraform would consider it computed by NSX)